src/main/java/uk/gov/ons/ctp/response/collection/exercise/lib/common/redis/EndpointCacheDataDeleter.java

```diff
@@ -0,0 +1,29 @@
+package uk.gov.ons.ctp.response.collection.exercise.lib.common.redis;
+
+import com.godaddy.logging.Logger;
+import com.godaddy.logging.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+@Component
+public class EndpointCacheDataDeleter {
+  /**
+   * This key was generated by caching the result of
+   * /collectionexercises/survey/{survey_id}?liveOnly=true in frontstage
+   */
+  private static final Logger log = LoggerFactory.getLogger(EndpointCacheDataDeleter.class);
+
+  private static final String REDIS_KEY_PREFIX = "frontstage:collection-exercise-by-survey-id:";
+
+  @Autowired private RedisUtil<Object> redisUtil;
+
+  public void deleteFrontstageExerciseByIdKey(String surveyId) {
+    String key = REDIS_KEY_PREFIX + surveyId;
+    try {
+      Boolean cacheInvalidated = redisUtil.deleteValue(key);
+      log.with("key", key).with("success", cacheInvalidated).info("Cache invalidated with key");
+    } catch (Exception e) {
+      log.with("key", key).error("Unable to remove key from redis", e);
```

Comment:
Could you put a comment here saying something like 'Not bubbling up the exception intentionally as a failed key removal isn't fatal and it has a ttl so it will expire naturally'